src/main/java/de/eldoria/pickmeup/listener/CarryListener.java

```diff
@@ -92,57 +92,113 @@ public void onEntityInteract(PlayerInteractAtEntityEvent event) {
-            return;
+                if (player.isSneaking() && mountStates.get(playerUUID) == MountState.SNEAK_THROW) {
+                    throwBarHandler.register(player);
+                }
+            }, 10);
         }
+    }
 
-        if (!event.isSneaking() && mountState == MountState.SNEAK_THROW) {
-            if (!throwBarHandler.isRegistered(player)) {
-                unmountAll(player);
-            } else {
-                double force = throwBarHandler.getAndRemove(player);
-                Vector throwVec = player.getEyeLocation().getDirection().normalize().multiply(force * config.carrySettings().throwForce());
-                player.getWorld().playSound(player.getLocation(), Sound.ENTITY_ENDER_DRAGON_FLAP, 1, 1);
-                for (Entity passenger : player.getPassengers()) {
-                    delayedActions.schedule(() -> trailHandler.startTrail(passenger), 2);
-                    player.removePassenger(passenger);
-                    passenger.setVelocity(throwVec);
-                    plugin.getLogger().config("Throwing entity | Location:" + player.getLocation().toVector()
-                                              + " | Force: " + force
-                                              + " | ThrowForce: " + config.carrySettings().throwForce()
-                                              + " | ViewVec: " + throwVec);
+    private void performThrow(Player player, UUID playerUUID) {
+        double force = throwBarHandler.getAndRemove(player);
+        Vector throwVec = player.getEyeLocation().getDirection().normalize().multiply(force * config.carrySettings().throwForce());
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_ENDER_DRAGON_FLAP, 1, 1);
+
+        List<Entity> passengers = new ArrayList<>(player.getPassengers());
+        for (Entity passenger : passengers) {
+            if (passenger == null || !passenger.isValid() || passenger.isDead()) {
+                player.removePassenger(passenger);
+                blocked.remove(passenger.getUniqueId());
+                continue;
+            }
+
+            delayedActions.schedule(() -> {
+                if (passenger.isValid() && !passenger.isDead()) {
+                    trailHandler.startTrail(passenger);
                 }
+            }, 2);
+
+            player.removePassenger(passenger);
+            passenger.setVelocity(throwVec);
+
+            plugin.getLogger().config("Throwing entity | Location:" + player.getLocation().toVector()
+                    + " | Force: " + force
+                    + " | ThrowForce: " + config.carrySettings().throwForce()
+                    + " | ViewVec: " + throwVec);
+        }
+        mountStates.remove(playerUUID);
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent event) {
+        Player player = event.getEntity();
+        UUID playerUUID = player.getUniqueId();
+        mountStates.remove(playerUUID);
+        throwBarHandler.remove(player);
+    }
```

Comment:
No need for bloating the method with declarations for single use.
```suggestion
    public void onPlayerDeath(PlayerDeathEvent event) {
        mountStates.remove(event.getEntity().getUniqueId());
        throwBarHandler.remove(event.getEntity());
    }
```